File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int red = 0;
        int white = 0;
        int blue = nums.length - 1;

        while (white <= blue) {
            if (nums[white] == 0) {
                int tmp = nums[red];
                nums[red++] = nums[white];
                nums[white++] = tmp;
            } else if (nums[white] == 1) {
                white++;
            } else {
                int tmp = nums[white];
                nums[white] = nums[blue];
                nums[blue--] = tmp;
            }
        }
    }
}

File: 0075-sort-colors/0075-sort-colors.kt
class Solution {
    fun sortColors(nums: IntArray): Unit {
        var red = 0
        var white = 0
        var blue = nums.lastIndex

        while (white <= blue) {
            when (nums[white]) {
                0 -> {
                    nums[red] = nums[white].also { nums[white] = nums[red] }
                    red++
                    white++
                }
                1 -> white++
                2 -> {
                    nums[white] = nums[blue].also { nums[blue] = nums[white] }
                    blue--
                }
            }
        }
    }
}